src/babylon.scene.ts

```diff
@@ -1924,6 +1924,31 @@
             this.onAfterRenderObservable.removeCallback(func);
         }
 
+        private _executeOnceBeforeRender(func: () => void): void {
+            let execFunc = () => {
+                func();
+                this.unregisterBeforeRender(execFunc);

```

Comment:
I recommend using a setTimeout(0) for the unregister as you are updating the list of callbacks inside a loop thought the same callback list